python/ray/tune/utils/util.py

```diff
@@ -298,6 +298,32 @@ def unflatten_dict(dt, delimiter="/"):
     return out
 
 
+def unflatten_list_dict(dt, delimiter="/"):
+    """Unflatten nested dict and list."""
+    out_type = list if list(dt)[0].split(delimiter, 1)[0].isdigit() else dict
+    out = out_type()
+    for key, val in dt.items():
+        path = key.split(delimiter)
+
+        item = out
+        for i, k in enumerate(path[:-1]):
+            next_type = list if path[i+1].isdigit() else dict
+            if isinstance(item, dict):
+                item = item.setdefault(k, next_type())
+            elif isinstance(item, list):
```

Comment:
```suggestion
            elif isinstance(item, (list, tuple)):
```

Reply:
First of all, `tuple` object cannot `append`.

Second, `item` won't be `tuple` since `out_type` and `next_type` are always `list` or `dict`.

So in my opinion this change isn't good. 😢 If I didn't get your point, please let me know :)

Reply:
you're right, I forgot that we're appending to `item`